awscli/customizations/configure/addmodel.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import json
+import os
+
+from botocore.model import ServiceModel
+
+from awscli.customizations.commands import BasicCommand
+
+
+def _get_endpoint_prefix_to_name_mappings(session):
+    # Get the mappings of endpoint prefixes to service names from the
+    # available service models.
+    prefixes_to_services = {}
+    for service_name in session.get_available_services():
+        service_model = session.get_service_model(service_name)
```

Comment:
I talked with you about this. Decided to check for its existence in ``session.get_available_services()``, then do the mapping creation if it does not exist. Here are some stats I got from running on my machine:

```
$ time aws configure add-model --service-model file://botocore/data/ec2/2015-10-01/service-2.json 

real	0m0.528s
user	0m0.329s
sys	0m0.119s

$ time aws configure add-model --service-model file://botocore/data/cloudwatch/2010-08-01/service-2.json

real	0m2.515s
user	0m1.707s
sys	0m0.226s

$ time aws configure add-model --service-model file://botocore/data/cloudwatch/2010-08-01/service-2.json --service-name cloudwatch

real	0m0.420s
user	0m0.308s
sys	0m0.110s
```
Based on these, I think this approach should be good to go with.